build-tools/automation/azure-pipelines.yaml

```diff
@@ -0,0 +1,128 @@
+# Java.Interop Pipelines
+
+trigger:
+  - master
+  - d16-*
+
+# Global variables
+variables:
+  DotNetCoreVersion: 2.1.701
+  HostedMac: Hosted Mac Internal
+  HostedWinVS2019: Hosted Windows 2019 with VS2019
+
+stages:
+  # Build and test on Windows
+- stage: windows_build
+  displayName: Windows
+  pool: $(HostedWinVS2019)
+  jobs:
+  - job: windows_build
+    displayName: Build and Test
+    timeoutInMinutes: 20
+    workspace:
+      clean: all
+    steps:
+    - checkout: self
+      submodules: recursive
+
+    - task: UseDotNet@2
+      displayName: Use .NET Core $(DotNetCoreVersion)
+      inputs:
+        version: $(DotNetCoreVersion)
+    
+    - task: MSBuild@1
+      displayName: MSBuild Java.Interop.sln /t:Prepare
+      inputs:
+        solution: Java.Interop.sln
+        configuration: $(Build.Configuration)
+        msbuildArguments: /t:Prepare
+    
+    - task: MSBuild@1
+      displayName: MSBuild Java.Interop.sln
+      inputs:
+        solution: Java.Interop.sln
+        configuration: $(Build.Configuration)
+    
+    - task: MSBuild@1
+      displayName: MSBuild RunNUnitTests.targets
+      inputs:
+        solution: build-tools/scripts/RunNUnitTests.targets
+        configuration: $(Build.Configuration)
+        msbuildArguments: /p:TestAssembly="bin\Test$(Build.Configuration)\generator-Tests.dll;bin\Test$(Build.Configuration)\Java.Interop.Tools.JavaCallableWrappers-Tests.dll;bin\Test$(Build.Configuration)\LogcatParse-Tests.dll;bin\Test$(Build.Configuration)\Xamarin.Android.Tools.ApiXmlAdjuster-Tests.dll;bin\Test$(Build.Configuration)\Xamarin.Android.Tools.Bytecode-Tests.dll"
+      condition: succeededOrFailed()
+    
+    - task: PublishTestResults@2
+      displayName: Publish Test Results
+      inputs:
+        testResultsFormat: NUnit
+        testResultsFiles: TestResult-*.xml
+      condition: succeededOrFailed()
+
+# Build and test on Mac
+- stage: mac_build
+  displayName: Mac
+  dependsOn: []
+  jobs:
+  - job: mac_build
```

Comment:
Same as above, we can fan out here based on Configuration as the existing build pipelines do.